src/error.rs

```diff
@@ -26,12 +27,20 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// [module documentation]: index.html
 ///
 /// *This type is available if Syn is built with the `"parsing"` feature.*
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct Error {
-    span: Span,
+    // Span is implemented as an index into a thread-local interner to keep the
+    // size small. It is not safe to access from a different thread. We want
+    // errors to be Send and Sync to play nicely with the Failure crate, so pin
+    // the span we're given to its original thread and assume it is
+    // Span::call_site if accessed from any other thread.
```

Comment:
> assume it is `Span::call_site` if accessed from any other thread.

Wouldn't this produce worse error messages when passed to another thread ?

Reply:
Yes it will. The compiler's source locations are all confined to one thread so there is no way around that.

Reply:
Can this PR also modify the docs of the `Error` type with this information ?